### PR TITLE
[IMP] Remove jQuery UI autocomplete

### PR DESCRIPTION
**Pull request by res-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/static/src/components/autocomplete_with_pages/url_autocomplete.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    get dropdownClass() {
+        if (this.props.options?.classes) {
+            const classList = [];
+            for (const key in this.props.options.classes) {
+                classList.push(key, this.props.options.classes[key]);
+            }
+            return classList.join(" ");
+        } else {
+            return "";
+        }
```
Comment (rfr-odoo) [inline addons/website/static/src/components/autocomplete_with_pages/url_autocomplete.js:35]:
```patch
     }
 
     get dropdownClass() {
-        if (this.props.options?.classes) {
-            const classList = [];
+        const classList = [];
+        if (this.props.options?.classes) {
             for (const key in this.props.options.classes) {
                 classList.push(key, this.props.options.classes[key]);
             }
-            return classList.join(" ");
-        } else {
-            return "";
-        }
+        }
+        return classList.join(" ");
     }
 
     get sources() {
```

Avoid multiple return

File: addons/website/static/tests/tours/edit_link_popover.js [hunk 2]
```diff
@@ -65,10 +65,16 @@ wTourUtils.registerWebsitePreviewTour('edit_link_popover', {
...
+    {
+        content: "Type the link URL /",
+        trigger: '#o_link_dialog_url_input',
+        run: "text /"
+    },
     ...clickFooter,
     {
         content: "Click on link",
         trigger: `${FIRST_PARAGRAPH} a`,
+        run: "click"
```
Comment (rfr-odoo) [inline addons/website/static/tests/tours/edit_link_popover.js:77]: Not needed as default action is click

File: addons/website/static/tests/tours/edit_link_popover.js [hunk 3]
```diff
@@ -93,6 +99,7 @@ wTourUtils.registerWebsitePreviewTour('edit_link_popover', {
     {
         content: "Click navbar menu Home",
         trigger: 'iframe #top_menu a:contains("Home")',
+        run: "click"
```
Comment (rfr-odoo) [inline addons/website/static/tests/tours/edit_link_popover.js:102]: Same as above

File: addons/website/static/tests/tours/edit_link_popover.js [hunk 4]
```diff
@@ -150,6 +157,7 @@ wTourUtils.registerWebsitePreviewTour('edit_link_popover', {
     {
         content: "Click 'Home' link in footer",
         trigger: 'iframe footer a[href="/"]',
+        run: "click"
```
Comment (rfr-odoo) [inline addons/website/static/tests/tours/edit_link_popover.js:160]: Same as above

File: addons/website/controllers/main.py [hunk 5]
```diff
@@ -366,10 +366,11 @@ def get_suggested_link(self, needle, limit=10):
         for name, url, mod in current_website.get_suggested_controllers():
             if needle.lower() in name.lower() or needle.lower() in url.lower():
                 module_sudo = mod and request.env.ref('base.module_%s' % mod, False).sudo()
-                icon = mod and "<img src='%s' width='24px' height='24px' class='mr-2 rounded' /> " % (module_sudo and module_sudo.icon or mod) or ''
+                icon = mod and '%s' % (module_sudo and module_sudo.icon or mod) or ''
```
Comment (rdeodoo) [inline addons/website/controllers/main.py:369]: Didn't test the PR but just tried this one and seems wrong:
Before:
image
After:
image

File: addons/website/static/src/components/autocomplete_with_pages/url_autocomplete.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
+/** @odoo-module **/
+
+import { Component } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+import { AutoCompleteWithPages } from "@website/components/autocomplete_with_pages/autocomplete_with_pages";
+
+export class UrlAutoComplete extends Component {
```
Comment (rdeodoo) [inline addons/website/static/src/components/autocomplete_with_pages/url_autocomplete.js:7]: Randomly commenting here:
image

The dropdown should be below the input, as before:
image


Even if it seems like we had a bug where the dropdown went offscreen (without possibility to scroll), which should be fixed indeed

File: addons/website/static/src/components/autocomplete_with_pages/url_autocomplete.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        if (!this.props.targetDropdown) {
+            this.props.onInput(...arguments);
+            return;
+        }
+        this.props.targetDropdown.value = inputValue;
+        this.props.options.urlChosen?.();
+    }
+}
+
+UrlAutoComplete.props = {
```
Comment (rdeodoo) [inline addons/website/static/src/components/autocomplete_with_pages/url_autocomplete.js:93]: Randomly commenting here: we lost the "tab" capability to select an URL in the url dropdown, can we make it work?
If not, it's not the end of the world I guess

image

File: addons/website/static/src/js/editor/snippets.options.js [hunk 8]
```diff
@@ -101,17 +101,25 @@ const UrlPickerUserValueWidget = InputUserValueWidget.extend({
         this.containerEl.after(linkButton);
         this.el.classList.add('o_we_large');
         this.inputEl.classList.add('text-start');
+    },
+
+    open() {
```
Comment (rdeodoo) [inline addons/website/static/src/js/editor/snippets.options.js:106]: Randomly commenting here, it's related to the snippet option but not this line
image

The url autocomplete in snippet option is not working anymore?

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Hello, quick one passing by thanks to the codeowner notification

Once those are fixed, could you have a look at all changed code occurrences functionally in the software to see if it's the expected result?
Then ping us so we can have a look too to approve the PR before it's r+?

File: addons/website/static/src/js/editor/snippets.options.js (hunk 8 above)
Comment (rdeodoo) [reply addons/website/static/src/js/editor/snippets.options.js:106]: Odoo 17: https://watch.screencastify.com/v/ohweJkaYACRfhYr3Tzph
Odoo 16.4: https://watch.screencastify.com/v/JORTqhJsx384RBZfKB40
Everything seems to be working fine in stable :shrug:

File: addons/website/static/src/client_actions/configurator/configurator.js [hunk 9]
```diff
@@ -92,55 +93,55 @@ Object.assign(WelcomeScreen, {
...
 
+    get sources() {
+        return [
+            {
+                options: (request) => {
+                    return request.length < 1 ? [] : this._autocompleteSearch(request);
+                },
+            },
+        ];
+    }
```
Comment (rdeodoo) [inline addons/website/static/src/client_actions/configurator/configurator.js:144]: This is to replace the `minLength: 1,` from jQuery, right?
Isn't that part of the owl `AutoComplete` implementation?

**Review by rdeodoo — APPROVED** [review]
Nice work.

Seems about right functionally, we are short on time to have a proper technical review.

File: addons/website/static/src/client_actions/configurator/configurator.js (hunk 9 above)
Comment (adr-odoo) [reply addons/website/static/src/client_actions/configurator/configurator.js:144]: By default, the owl `AutoComplete` doesn't have this limit. So it's 0 for now.

**Review by adr-odoo — APPROVED** [review]
@robodoo rebase-ff r+

**Comment by Zynton** [comment]
Congrats :tada:

**Pull request closed without merging**